### PR TITLE
fix(sync-actions): fix setCustomType on Categories

### LINKED ISSUE #320: Error setting custom Type on Category with no previous custom Type
### Description
SomeHow jsondiffpatch library is returning a diff that contains an array in the custom type field instead of an object.
Because of this, in `actionMapCustom` function when check 
 `diff.custom.type && diff.custom.type.id` always fails because this diff.custom.type is returning as an array

### PR DESCRIPTION
**Pull request by wizzy25**
affects: @commercetools/sync-actions
resolves #320 
#### Summary

fix setCustomType on Categories
#### Description

fix issue where the setCustomType action could not be properly
built on categories with no prior
custom object

### CODE REVIEW

File: packages/sync-actions/test/category-sync.spec.js [hunk 1]
```diff
@@ -75,6 +75,72 @@ describe('Actions', () => {
...
+          id: 'customType2',
+        },
+        fields: {
+          customField1: true,
+        },
+      },
+    }
+    const now = {
+      key: 'category-key',
+      custom: {},
```
[the remark this answers is not in the document]
Comment (wizzy25) [reply packages/sync-actions/test/category-sync.spec.js]: Yes. see [here](https://github.com/commercetools/nodejs/pull/336/files/472ae5794015a99a76d12edae2dec3e9f2192fbe#diff-5addf3aceba9e697c10d46bddebf8a16R79)

**Comment by wizzy25** [comment]
With the current implementation, if the old `category` contains a `custom` object and the new `category` contains an empty `custom` object or does not contain a `custom` object, it will be unset in both scenarios. Is this a desired behaviour?

**Comment by junajan** [comment]
@wizzy25 so the question is whether we want to do a full change or a diff change only? Not sure how to answer :-X

File: packages/sync-actions/test/category-sync.spec.js [hunk 2]
```diff
@@ -75,6 +75,93 @@ describe('Actions', () => {
     expect(actual).toEqual(expected)
   })
 
+  it('build `setCustomType` action on category with no `custom` object', () => {
```
Comment (PhilippSpo) [inline packages/sync-actions/test/category-sync.spec.js]: All of these its have the `build `setCustomType` action on category` part in common. Thus we could wrap all of them in a describe block:
```js
describe('changing the custom type of a category', () => {
  describe('existing category has no `custom` object', () => {
    it('should build a `setCustomType` action with the new type')
  })
  describe('existing category has an empty `custom` object', () => {
    it('should build a `setCustomType` action with the new type')
  })
  describe('existing category has a `custom` object', () => {
    describe('new category has no `custom` object', () => {
      it('should build a `setCustomType` action to unset the `custom` type')
    })
    describe('new category has an empty `custom` object', () => {
      it('should build a `setCustomType` action to unset the `custom` type')
    })
  })
})
```

**Pull request merged**